src/libstd/io/stdio.rs

```diff
@@ -872,6 +874,82 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that a prompt
+/// is printed before capturing input. This function returns
+/// either success ([`Ok`]) or failure ([`Err`]) containing
+/// the input. See the [`io::Result`] module documentation
+/// for details.
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(prompt)]
+/// use std::io::prompt;
+///
+/// fn main() {
+///     let name = prompt("Enter name: ").expect("input failed!");
+///
+///     println!("Your name is {}!", name);
+/// }
+/// ```
+#[unstable(
+    feature = "prompt",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
```

Comment:
```suggestion
    reason = "this function may be replaced with a more general mechanism",
```
as suggested by @GuillaumeGomez